lib/mlir/Dialect/Clift/IR/CliftAttributes.cpp

```diff
@@ -246,11 +284,8 @@ FunctionTypeAttr::verify(EmitErrorType EmitError,
                          llvm::StringRef Name,
                          clift::ValueType ReturnType,
                          llvm::ArrayRef<FunctionArgumentAttr> Args) {
-  if (const auto Type = mlir::dyn_cast<DefinedType>(ReturnType)) {
-    if (mlir::isa<FunctionTypeAttr>(Type.getElementType()))
-      return EmitError() << "function type cannot return another function type";
-  }
-
+  if (not isVoid(ReturnType) and not isObjectType(ReturnType))
+    return EmitError() << "Function return type must be an object type or void";
```

Comment:
:thinking: this makes it impossible to have return types fields that are primitives, enums, pointers, which is wrong.

The only forbidden types should be incomplete types and function types.